_docs/concepts/auth.md

```diff
@@ -0,0 +1,138 @@
+
+#### Key Management {#key-management}
+
+Istio Auth provides a per-cluster CA (Certificate Authority) to automate key & cert management. It mainly performs 4 key operations:
+
+
+
+*   Generate a key/cert pair for each service account.
+*   Distribute key/cert to each pod according to the service account.
+*   Rotate key/cert periodically. 
+*   Revoke a specific key/cert pair when necessary.
+
+### Workflow 
+
+Istio Auth workflow consists of two phases, deployment and runtime. We briefly cover each phase in this section and a more detailed version can be found [here](https://docs.google.com/a/google.com/document/d/1spoQ9MIb7ABFDdFzlFITczCbH_AHO3RXSgLLeXAYIJU/edit?usp=sharing).
+
+#### Deployment Phase {#deployment-phase}
+
+
+
+1.  Istio CA watches K8s API Server, creates a cert/key pair for each of the existing and new service accounts, and sends them to API Server. 
+1.  When a pod is created, API Server mounts the cert/key according to the service account using [kubernetes secrets](https://kubernetes.io/docs/concepts/configuration/secret/).
+1.  [Istio Manager](https://github.com/istio/manager/blob/master/doc/design.md) generates the config with proper cert/key and secure naming information, which defines what service account(s) can run a certain service, and passes it to Envoy. 
+
+#### Runtime Phase {#runtime-phase}
+
+
+
+1.  The outbound traffic from a client service is rerouted to its local Envoy. 
+1.  The client side Envoy starts mutual TLS handshake with the server side Envoy. During the handshake, it also does secure naming check to verify that the service account presented in the server certificate can run the server service. 
+1.  The traffic is forwarded to the server side Envoy after mTLS connection is established, which is then forwarded to the server service through local TCP connections.
+
+### Service to Service Auth Best Practice {#service-to-service-auth-best-practice}
+
+In this section, we provide a few deployment guidelines and then discuss a real-world scenario. 
+
+#### Deployment Guidelines {#deployment-guidelines}
+
+
+
+*   If there are multiple service operators (a.k.a. [SREs](https://en.wikipedia.org/wiki/Site_reliability_engineering)) deploying different services in a cluster (typically in a medium- or large-size cluster), we recommend to create a separate [namespace](https://kubernetes.io/docs/tasks/administer-cluster/namespaces-walkthrough/) for each SRE team to isolate their access. For example, we can create a "team1-ns" namespace for team1, and "team2-ns" namespace for team2, such that both teams won't be able to access each other's services.
+*   If Istio CA is compromised, all its managed key & cert in the cluster may be exposed. We *strongly *recommend to run Istio CA on a dedicated namespace (e.g., istio-ca-ns) which only cluster admins have access to.
```

Comment:
All its managed keys and certs

*strongly * -> *strongly* (remove extra space)

recommend running Istio CA on a dedicated namespace (for example,  istio-ca-ns), which only...